components/pd_client/src/util.rs

```diff
@@ -242,11 +242,11 @@ where
     Resp: Send + 'static,
     F: FnMut(&RwLock<Inner>, Req) -> PdFuture<Resp> + Send + 'static,
 {
-    async fn reconnect_if_needed(&mut self) -> bool {
+    async fn reconnect(&mut self) -> bool {
```

Comment:
emm... personally i think it better to keep that `if_needed`, because the early return in L249 means "not needed".